test/unit/Optimize.ml

```diff
@@ -429,6 +446,47 @@ let%expect_test "do not inline recursive functions" =
         if(PNot__(emit_generated_quantities__)) return;
       } |}]
 
+let%expect_test "do not try to inline extern functions" =
+  let before = !Frontend.Typechecker.check_that_all_functions_have_definition in
+  Frontend.Typechecker.check_that_all_functions_have_definition := false ;
+  let mir =
+    reset_and_mir_of_string
+      {|
+            functions {
+              int fib(int n);
+            }
+            model {
+              reject(fib(5));
+            }
+            |}
+  in
+  Frontend.Typechecker.check_that_all_functions_have_definition := before ;
+  let mir = function_inlining mir in
+  Fmt.str "@[<v>%a@]" Program.Typed.pp mir |> print_endline ;
+  [%expect
+    {|
+            functions {
+              int fib(int n) {
+                ;
+              }
```

Comment:
This looks a bit weird to me. Might want to change `Middle.Program.pp_fun_def`? (That function should be using `UnsizedType.pp_returntype` anyway...)
How do you feel about, say
```
extern int fib(int n);
```